packages/gestalt/src/SheetMobile/FullPage.js

```diff
@@ -61,8 +62,25 @@ export default function FullPage({
   showDismissButton,
   subHeading,
 }: Props): Node {
+  const id = useId();
+
+  // Consumes DefaultLabelProvider
   const { accessibilityLabel: defaultAccessibilityLabel } = useDefaultLabelContext('SheetMobile');
 
+  // Consumes GlobalEventsHandlerProvider
+  const { sheetMobile } = useGlobalEventsHandlerContext() ?? {
+    sheetMobile: { onOpen: () => {}, onClose: () => {} },
+  };
+
+  useEffect(() => {
+    sheetMobile?.onOpen?.();
+
+    return function cleanup() {
+      sheetMobile?.onClose?.();
+    };
+  }, [sheetMobile]);
```

Comment:
Passing an object to `useEffect` isn't great, as the object reference can stay the same while its properties change, and the effect won't run again. We should destructure the handlers out of `sheetMobileHandlers` and use those directly in the dependency array.

Reply:
DONE